apps/wizard/app_pages/indicator_upgrade/dataset_selection.py

```diff
@@ -145,10 +196,9 @@ def build_dataset_form(df: pd.DataFrame, similarity_names: Dict[str, Any]) -> "S
         )
         reset_indicator_form()
 
-    # Get IDs of datasets
-    dataset_old_id = display_name_to_id_mapping[dataset_old]
-    dataset_new_id = display_name_to_id_mapping[dataset_new]
-    set_states({"migration_new_id": dataset_new_id})
+    # Get IDs of the chosen old and new datasets.
+    dataset_old_id = display_name_to_id_mapping.get(dataset_old) or display_step_to_id_mapping.get(dataset_old)
+    dataset_new_id = display_name_to_id_mapping.get(dataset_new) or display_step_to_id_mapping.get(dataset_new)
```

Comment:
instead, you could use:

```py
dataset_old_id = display_name_to_id_mapping.get(dataset_old, display_step_to_id_mapping.get(dataset_old))
dataset_new_id = display_name_to_id_mapping.get(dataset_new, display_step_to_id_mapping.get(dataset_new))

Reply:
Thanks, but I think the current option is clearer.